### PR TITLE
Pastix: new package

### PR DESCRIPTION
**Pull request by fpruvost**
PaStiX (Parallel Sparse matriX package) is a scientific library that provides a
high performance parallel solver for very large sparse linear systems based on
direct methods.

### CODE REVIEW

**Review by mfaverge — DISMISSED** [review]

File: var/spack/repos/builtin/packages/pastix/package.py
Comment (ramet) [inline var/spack/repos/builtin/packages/pastix/package.py]: We should never link with a multi-threaded BLAS.

File: var/spack/repos/builtin/packages/pastix/package.py
Comment (ramet) [inline var/spack/repos/builtin/packages/pastix/package.py:79]: Here again, OpenBLAS has to be built with a sequential version.

File: var/spack/repos/builtin/packages/pastix/package.py
Comment (fpruvost) [reply var/spack/repos/builtin/packages/pastix/package.py]: Ok done

File: var/spack/repos/builtin/packages/pastix/package.py
Comment (fpruvost) [reply var/spack/repos/builtin/packages/pastix/package.py:79]: This is the case by default in spack, openblas is sequential (variant threads=none). It is not possible to force it at the cmake  level because the library filename is the same (seq. vs. multithreaded).

Comment (mfaverge) [reply var/spack/repos/builtin/packages/pastix/package.py:79]: In fact threads=None is not enough for OpenBLAS as the lib is not reentrant. 
If you look at the packages from Debian/ubuntu for example, what is important is the fact that OpenBLAS has been built with `USE_OPENMP=1`, but I don't know how to check if it's the case or not.

Comment (fpruvost) [reply var/spack/repos/builtin/packages/pastix/package.py:79]: I have added the following to fix this `depends_on('openblas threads=openmp', when='^openblas')`

**Review by mfaverge — APPROVED** [review]

**Review by ramet — APPROVED** [review]

**Review by mfaverge — APPROVED** [review]

**Review by mfaverge — APPROVED** [review]

**Comment by fpruvost** [comment]
@tldahlgren reviews have been done from our side. Now, a reviewer with write access is necessary. Is it possible to get a review from the spack core developers ?
Thanks

**Review by tldahlgren — APPROVED** [review]
LGTM. Confirmed sha56.

**Pull request merged**
